src/net_processing.cpp

```diff
@@ -92,6 +94,7 @@ struct COrphanTx {
 };
 RecursiveMutex g_cs_orphans;
 std::map<uint256, COrphanTx> mapOrphanTransactions GUARDED_BY(g_cs_orphans);
+std::map<uint256, std::map<uint256, COrphanTx>::iterator> g_orphans_by_wtxid GUARDED_BY(g_cs_orphans);
```

Comment:
You should probably wipe this in ~CNetProcessingCleanup() like we do the other orphan fields, but, also, why the hell is that thing even there...

Reply:
Fixed, thanks.

Not sure exactly what you're asking but it seemed easier to add an additional map, rather than change the existing data structure to support wtxid-lookup.